Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs

```diff
@@ -100,5 +104,133 @@ public void ApplyEnabledStateCleansUp()
                 Assert.IsTrue(removeErrorLogFilesCalled);
             }
         }
+
+        [TestMethod]
+        public void OnChannelGroupReadySendsPendingCrashes()
+        {
+            var expectedFileInfo1 = new FileInfo("file");
+            var expectedFileInfo2 = new FileInfo("file2");
+            var expectedprocessId = 123;
+            var expectedLogIds = new List<Guid>();
+            var removedLogIds = new List<Guid>();
+            using (ShimsContext.Create())
+            {
+                // Stub get/read/delete error files
+                ShimErrorLogHelper.GetErrorLogFiles = () => new List<FileInfo> { expectedFileInfo1, expectedFileInfo2 };
+                ShimErrorLogHelper.RemoveStoredErrorLogFileGuid = (Guid guid) => removedLogIds.Add(guid);
+                ShimErrorLogHelper.ReadErrorLogFileFileInfo = (FileInfo file) =>
+                {
+                    var errorLog = new ManagedErrorLog
+                    {
+                        Id = Guid.NewGuid(),
+                        ProcessId = expectedprocessId
+                    };
+                    expectedLogIds.Add(errorLog.Id);
+                    return errorLog;
+                };
+
+                Crashes.SetEnabledAsync(true).Wait();
+                Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+
+                // Verify crashes logs have been queued to the channel
+                foreach (var expectedLogId in expectedLogIds)
+                {
+                    _mockChannel.Verify(channel => channel.EnqueueAsync(It.Is<ManagedErrorLog>(log => log.Id == expectedLogId && log.ProcessId == expectedprocessId)), Times.Once());
+                }
+                CollectionAssert.AreEqual(expectedLogIds, removedLogIds);
+            }
+        }
+
+        [TestMethod]
+        public void OnChannelGroupReadyDoesNotSendsPendingCrashesOnDisabled()
+        {
+            using (ShimsContext.Create())
+            {
+                var getErrorLogFilesCalled = false;
+                // Stub get/read/delete error files
+                ShimErrorLogHelper.GetErrorLogFiles = () =>
+                {
+                    getErrorLogFilesCalled = true;
+                    return new List<FileInfo>();
+                };
+                Crashes.SetEnabledAsync(false).Wait();
+                Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+                _mockChannel.Verify(channel => channel.EnqueueAsync(It.IsAny<ManagedErrorLog>()), Times.Never());
+                Assert.IsFalse(getErrorLogFilesCalled);
```

Comment:
Is there any positive verification we can do here?  This verification is all "this stuff didn't happen", is there anything that we can verify _did_ happen?

Reply:
There is a test to verify the behavior when it is enabled but the test doesn't use same verification since it has clear way to verify it using channel.